src/gui/speedlimitdialog.cpp

```diff
@@ -50,58 +92,85 @@ SpeedLimitDialog::~SpeedLimitDialog()
     delete m_ui;
 }
 
-// -2: if cancel
-long SpeedLimitDialog::askSpeedLimit(QWidget *parent, bool *ok, const QString &title, const long defaultVal, const long maxVal)
+void SpeedLimitDialog::accept()
 {
-    if (ok) *ok = false;
+    const int uploadLimit = (getUploadSpeedLimit() * 1024);
+    const int downloadLimit = (getDownloadSpeedLimit() * 1024);
+    const int altUploadLimit = (getAltUploadSpeedLimit() * 1024);
+    const int altDownloadLimit = (getAltDownloadSpeedLimit() * 1024);
 
-    SpeedLimitDialog dlg(parent);
-    dlg.setWindowTitle(title);
-    dlg.setupDialog((maxVal / 1024.), (defaultVal / 1024.));
+    BitTorrent::Session::instance()->setGlobalUploadSpeedLimit(uploadLimit);
+    BitTorrent::Session::instance()->setGlobalDownloadSpeedLimit(downloadLimit);
+    BitTorrent::Session::instance()->setAltGlobalUploadSpeedLimit(altUploadLimit);
+    BitTorrent::Session::instance()->setAltGlobalDownloadSpeedLimit(altDownloadLimit);
 
-    if (dlg.exec() == QDialog::Accepted)
-    {
-        if (ok) *ok = true;
+    QDialog::accept();
+}
 
-        const int val = dlg.getSpeedLimit();
-        if (val < 0)
-            return 0;
-        return (val * 1024);
-    }
+void SpeedLimitDialog::updateDownloadSpinValue(const int value)
+{
+    m_ui->spinDownloadLimit->setValue(value);
+}
 
-    return -2;
+void SpeedLimitDialog::updateUploadSpinValue(const int value)
+{
+    m_ui->spinUploadLimit->setValue(value);
 }
 
-void SpeedLimitDialog::updateSpinValue(const int value)
+void SpeedLimitDialog::updateAltDownloadSpinValue(const int value)
 {
-    m_ui->spinBandwidth->setValue(value);
+    m_ui->spinAltDownloadLimit->setValue(value);
 }
 
-void SpeedLimitDialog::updateSliderValue(const int value)
+void SpeedLimitDialog::updateAltUploadSpinValue(const int value)
 {
-    if (value > m_ui->bandwidthSlider->maximum())
-        m_ui->bandwidthSlider->setMaximum(value);
-    m_ui->bandwidthSlider->setValue(value);
+    m_ui->spinAltUploadLimit->setValue(value);
 }
 
-int SpeedLimitDialog::getSpeedLimit() const
+void SpeedLimitDialog::updateDownloadSliderValue(const int value)
```

Comment:
Those update functions has the same pattern so it would be recommended to abstract it:
```c++
namespace 
{
    void updateSliderValue(QSlider *slider, const int value)
    {
        if (value > slider->maximum())
            slider->setMaximum(value);
        slider->setValue(value);
    }
}

void SpeedLimitDialog::updateDownloadSliderValue(const int value)
{
    updateSliderValue(m_ui->sliderDownloadLimit, value);
}

// or just where you need it:
connect(m_ui->spinDownloadLimit, qOverload<int>(&QSpinBox::valueChanged)
            , this, [this](const int value) { updateSliderValue(m_ui->sliderDownloadLimit, value); });
```

Reply:
I chose the first, thanks.